### PR TITLE
Disables the Jetpack No Sites error for logins and signups

### PR DESCRIPTION
**Pull request by emilylaguna**
Since we're allowing site creation we shouldn't display the no sites error if the user logs in or signs up. 

This also fixes a small issue where the tapping on the 'Go to Reader' button from the Notifications.

### To test:
1. Follow the steps from here: https://github.com/wordpress-mobile/WordPress-iOS/pull/17914

#### No Sites Login
1. Open the app
2. Tap the Continue with WP.com button
3. Login to an account that does not have any linked sites
4. Verify you are brought to the rest of the login flow and not displayed an error

#### Signup
1. Open the app
2. Tap the Continue with WP.com button
3. Create a user account
4. Verify you are brought to the rest of the onboarding flow and not displayed an error

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by emilylaguna** [comment]
cc @ashiagr

This PR disables the no sites error message for Jetpack app now, thanks for bringing this up!

**Review by momo-ozawa — COMMENTED** [review]
Works as described. 

One small thing. I noticed the asset we use in the PostSignupInterstitialVC is blue instead of green. Maybe we can reuse `jetpack-empty-state-illustration`.


Screen Shot 2022-02-14 at 9 52 09

**Comment by momo-ozawa** [comment]
@emilylaguna One other thing -- we should update the release notes. (Thanks for the reminder @ashiagr!)

**Comment by emilylaguna** [comment]
@momo-ozawa I updated the image, and the release notes. Can I get another review please?

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -3,6 +3,7 @@
 * [*] Site previews: Reduced visual flickering when previewing sites and templates. [#17861]
 * [*] Stats: fix navigation between Stats tab. [#17894]
 * [*] Add "Copy Link" functionality to Posts List and Pages List [#17911]
+* [*] [Jetpack-only] Enables the ability to use and create WordPress.com sites, and enables the Reader tab.
```
Comment (momo-ozawa) [inline RELEASE-NOTES.txt]:
```suggestion
* [*] [Jetpack-only] Enables the ability to use and create WordPress.com sites, and enables the Reader tab. [#17914, #17948]
```

**Review by momo-ozawa — APPROVED** [review]
The image looks good! Left 1 minor suggestion.

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (emilylaguna) [reply RELEASE-NOTES.txt]: 🤦‍♀️ Thanks, I should have added those.

**Pull request merged**